hermes/crates/hermes-ipfs/src/lib.rs

```diff
@@ -0,0 +1,273 @@
+//! Hermes IPFS
+//!
+//! Provides support for storage, and `PubSub` functionality.
+
+use std::{
+    ops::{Deref, DerefMut},
+    str::FromStr,
+};
+
+/// IPFS Content Identifier.
+pub use libipld::{self as ipld, Cid};
+/// Peer Info type.
+pub use rust_ipfs::p2p::PeerInfo;
+/// Enum for specifying paths in IPFS.
+pub use rust_ipfs::path::IpfsPath;
+/// Multiaddr type.
+pub use rust_ipfs::Multiaddr;
+/// Peer ID type.
+pub use rust_ipfs::PeerId;
+/// Builder type for IPFS Node configuration.
+pub use rust_ipfs::UninitializedIpfsNoop as IpfsBuilder;
+use rust_ipfs::{unixfs::AddOpt, Ipfs};
+
+/// Hermes IPFS Node
+///
+/// Provides the functionality of the inner `IPFS` by de-referencing.
+#[derive(Clone, Debug)]
+pub struct Node(Ipfs);
+
+impl From<Ipfs> for Node {
+    fn from(value: Ipfs) -> Self {
+        Self(value)
+    }
+}
+
+impl Deref for Node {
+    type Target = Ipfs;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Node {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// Hermes IPFS
+#[allow(dead_code)]
+pub struct HermesIpfs {
+    /// IPFS node
+    pub node: Node,
```

Comment:
What is the purpose to make this field public ? Seems like it should a private.
Also if change public to private for this field, seems like make sense also get rid of the `Node` struct and use just a plain `Ipfs` type here.

Reply:
The purpose is to have access to low-level functionality that the inner node provides. This is useful for more complex use cases, as can be seen in `examples/pubsub.rs`. I think it's preferrable to have our new type for this, and not re-use Ipfs directly, as we might want to swap the implementation in the future.

Reply:
I see your point.
But from my point of view we are trying to give for the user too much flexibility and responsibility.
Actually deferring the choice how it should be used.
I think we should hide such internal details, just to be safe. Also current interface looks exhaustive already and even if we will need something we can easily extend it.
But I dont have a really strong position here, I just feel that it should be private 😊

Reply:
I can't think of any reason to defend it being public. I've updated the code.